docs/allocation.md

```diff
@@ -0,0 +1,152 @@
+reach the maximum requested TLH size more quickly.
+
+## Heap configuration
+
+Depending on the memory management strategy in force, the Java heap can be configured in a number of ways. The simplest configuration consists of a single area of memory, often referred to as a *flat* heap. Other configurations divide the heap into different *areas* or *regions*, which might contain objects of different ages (generations) or sizes.
+
+### Area-based heaps
+
+The default GC policy for OpenJ9 uses a heap configuration that is divided into two main areas: the *nursery* area for new object allocation, and the *tenure* area for objects that continue to be reachable for a longer period of time. Most objects have short lifecycles and can be reclaimed by the garbage collector more quickly by focusing only on the nursery area. Global GC cycles that cause application pauses in order to clear and defragment the *tenure* area are less frequent.
+
+### SOA and LOA
+
+Some GC policies subdivide a *flat* heap or an *area* of the heap for object allocation into the Small Object Area (SOA) and the Large Object Area (LOA).
+
+The allocator initially attempts to allocate objects in the SOA, regardless of size. If the allocation cannot be satisfied the following actions are possible, depending on object size:
+
+- If the object is smaller than 64 KB, an allocation failure occurs, which triggers a GC action.
+- If the object is larger than 64 KB, the allocator attempts to allocate the object in the LOA. If the allocation cannot be satisfied, an allocation failure occurs, which triggers a GC action.
+
+The GC action that is triggered by the allocation failure depends on the GC policy in force.
+
+The overall size of the LOA is calculated when the heap is initialized, and recalculated at the end of each global GC cycle. The GC can expand or shrink the LOA, depending on usage, to avoid allocation failures.
+
+You can control the size of the LOA by using the [`-Xloainitial`, `-Xloaminimum`, and `-Xloamaximum`](xloaminimum.md) command line options. If the LOA is not used, the GC shrinks the LOA after a few cycles, down to the value of `-Xloaminimum`. You can also specify [`-Xnoloa`](xloa.md) to prevent an LOA being created.
+
+An SOA and LOA are used by the OpenJ9 GC policies: `gencon`, `optavgpause` and `optthruput`. For the `gencon` policy, the LOA and SOA are contained within the *tenure* area, which is designated for ageing objects. For more information about policies, see [Garbage collection policies](gc.md).
+
+### Region-based heaps
+
+The Java heap can also be subdivided into multiple regions. The `balanced` GC policy uses a heap that is divided into thousands of equal size regions in order to manage multiple generations of objects. The `metronome` GC policy also uses multiple regions, but each region is segregated by class size to manage a singe generation of objects. Both of these policies also use structures called *arraylets* to store large arrays in the heap.
```

Comment:
Some facts you may want to add

- region size is determined at JVM startup, and the size remains fixed throughout the run
- regions size is a power of 2
- for metronome, regions size defaults to 64KB, unless otherwise specified (by cmdline option)
- for balanced, regions size is a function of _maximum_ heap size (Xmx value), and is chosen such that there are 1024 and 2047 regions, except for very small heaps where there could be less regions, due to min region size of 512KB.
- for metronome, region count is fixed (heap in metronome is fully expanded from the start and never contracts)
- for balanced, actually number of active regions will depend of actual size of heap, but at most (when/if heap is fully expanded) there will be between 1024-2047 regions (as determined at startup)

Reply:
I'm not the originator of term 'segregate' (which indeed internally is used for metronome heap), but I think it's correct to say that 'heap is segregated (into subsets of regions of same size size-class)', not 'region is segregated'.

Reply:
>Some facts you may want to add

I think we capture most of these in the policy topic, but I'll double check and make sure we aren't missing anything. For this topic, I will add a link so that users can read more about the specifics for each policy.

Reply:
Double checked and all those facts are in the policy descriptions.